ios/FluentUI/Vnext/HUD/HeadsUpDisplay.swift

```diff
@@ -0,0 +1,227 @@
+            } else {
+                dismissAnimated()
+            }
+        })
+        .onAnimationComplete(for: presentationScaleFactor) {
+            resetScaleFactor()
+        }
+        .opacity(opacity)
+        .scaleEffect(presentationScaleFactor)
+        .onTapGesture {
+            state.tapAction?()
+        }
+    }
+
+    /// UIKit wrapper that exposes the SwiftUI Heads-up Display implementation.
+    /// - Parameters:
+    ///   - type: The `HUDType` enum value of the Heads-up Display.
+    ///   - isPresented: Binding boolean that controls whether the Heads-up Display is presented.
+    ///   - label: Label string presented by the Heads-up Display.
+    ///   - tapAction: Closure executed when the user taps on the Heads-up Display.
+    public init(type: HUDType,
+                isPresented: Binding<Bool>? = nil,
+                label: String? = nil,
+                tapAction: (() -> Void)? = nil) {
+        let stateImpl = MSFHUDStateImpl(type: type)
+        stateImpl.label = label
+        stateImpl.tapAction = tapAction
+        state = stateImpl
+
+        if let isPresented = isPresented {
+            _isPresented = isPresented
+            opacity = Constants.opacityDismissed
+            presentationScaleFactor = HUD.Constants.showAnimationScale
+        } else {
+            _isPresented = .constant(true)
+            opacity = Constants.opacityPresented
+            presentationScaleFactor = Constants.presentationScaleFactorDefault
+        }
+    }
+
+    let defaultTokens: HeadsUpDisplayTokens = .init()
+    var tokens: HeadsUpDisplayTokens {
+        return resolvedTokens
+    }
+    @Environment(\.fluentTheme) var fluentTheme: FluentTheme
+    @Binding var isPresented: Bool
+    @State var opacity: Double
+    @State var presentationScaleFactor: CGFloat
+    @ObservedObject var state: MSFHUDStateImpl
+
+    private func resetScaleFactor() {
+        guard presentationScaleFactor != Constants.presentationScaleFactorDefault else {
+            return
+        }
+
+        presentationScaleFactor = HUD.Constants.showAnimationScale
+    }
+
+    private func presentAnimated() {
+        withAnimation(.linear(duration: HUD.Constants.showAnimationDuration)) {
+            opacity = Constants.opacityPresented
+            presentationScaleFactor = Constants.presentationScaleFactorDefault
+        }
+    }
+
+    private func dismissAnimated() {
+        withAnimation(.linear(duration: HUD.Constants.hideAnimationDuration)) {
+            opacity = Constants.opacityDismissed
+            presentationScaleFactor = HUD.Constants.hideAnimationScale
+        }
+    }
+
+    private struct Constants {
+        static let presentationScaleFactorDefault: CGFloat = 1
+        static let opacityPresented: Double = 1
+        static let opacityDismissed: Double = 0
+    }
+}
+
+/// UIKit wrapper that exposes the SwiftUI Heads-up display implementation
+open class MSFHeadsUpDisplay: ControlHostingView {
```

Comment:
We've been putting these in separate files now to minimize the amount of UIKit knowledge needed in SwiftUI files. (Ideally we won't need to `import UIKit` at all in this file.)